orttraining/orttraining/training_api/include/module.h

```diff
@@ -111,7 +111,7 @@ struct Module {
   std::vector<OrtValue> weights_;
   std::vector<OrtValue> gradients_;
   bool accumulate_gradient_ = false;
-  const std::unordered_map<std::string, std::shared_ptr<Parameter>>& named_parameters_;
+  const std::unordered_map<std::string, std::shared_ptr<Parameter>> named_parameters_;
```

Comment:
Now that you are exposing Checkpoint State and it outlives Module class, do we need this change?